chanfitness/chaneventstore_testctx_test.go

```diff
@@ -0,0 +1,236 @@
+// been opened.
+func (c *chanEventStoreTestCtx) sendChannelOpenedUpdate(pubkey *btcec.PublicKey,
+	channel wire.OutPoint) {
+
+	update := channelnotifier.OpenChannelEvent{
+		Channel: &channeldb.OpenChannel{
+			FundingOutpoint: channel,
+			IdentityPub:     pubkey,
+		},
+	}
+
+	c.channelSubscription.sendUpdate(update)
+}
+
+// assertEvents asserts that a channel event store has an expected set of
+// events. This function can only be called once the test context's store has
+// been stopped.
+func (c *chanEventStoreTestCtx) assertEvents(channel wire.OutPoint,
+	events []channelEvent) {
+
+	// Fail if our store has not been stopped, because we will have data
+	// races if we try to assert a set of events while the store may still
+	// be updating them.
+	select {
+	case <-c.stopped:
+
+	default:
+		c.t.Fatalf("cannot assert events with an active store")
+	}
+
+	eventLog, ok := c.store.channels[channel]
+	require.True(c.t, ok, "channel not found")
+
+	// Check that our opened time is always non-zero, covering a
+	// previous bug where open times were not set for channels with
+	// peers who are offline when we add the channel to the store.
+	require.False(c.t, eventLog.openedAt.IsZero(), "zero open time")
+
+	// Assert that we have the number of events we expect.
+	actualEvents := eventLog.events
+	require.Len(c.t, actualEvents, len(events), "event count wrong")
+
+	// Run through our expected events and check that we have the
+	// right kind. We do not check time (at this stage) because it
+	// is not mocked.
+	for i, event := range events {
+		require.Equal(c.t, event.eventType, actualEvents[i].eventType,
+			"event type wrong")
+	}
+}
+
+// mockSubscription is a mock subscription client that blocks on sends into the
+// updates channel. We use this mock rather than an actual subscribe client
+// because they do not block, which makes tests race (because we have no way
+// to guarantee that the test client consumes the update before shutdown).
+type mockSubscription struct {
+	t       *testing.T
+	updates chan interface{}
+	quit    chan struct{}
+}
+
+// newMockSubscription creates a mock subscription.
+func newMockSubscription(t *testing.T) *mockSubscription {
+	return &mockSubscription{
+		t:       t,
+		updates: make(chan interface{}),
+		quit:    make(chan struct{}),
+	}
+}
+
+// sendUpdate sends an update into our updates channel, mocking the dispatch of
+// an update from a subscription server. This call will fail the test if the
+// update is not consumed within our timeout.
+func (m *mockSubscription) sendUpdate(update interface{}) {
+	select {
+	case m.updates <- update:
+
+	case <-time.After(timeout):
+		m.t.Fatalf("update: %v timeout", update)
+	}
+}
+
+// Updates returns the updates channel for the mock.
+func (m *mockSubscription) Updates() <-chan interface{} {
+	return m.updates
+}
+
+// Quit returns the mock's quit channel. We close this when we stop the mock.
+func (m *mockSubscription) Quit() <-chan struct{} {
+	return m.quit
```

Comment:
nice idea, added an assertion that the updates channel is closed. Ended up removing this function entirely and embedding the interface, since it's not used in the current set of tests.